### PR TITLE
Feature/set association

### PR DESCRIPTION
**Pull request by emarkees**
## In this milestone:

- The book.rb, classroom.rb, and rental.rb files were added to the project.
- Has-many/belongs-to relationship was performed between the class Classroom and Student.
- The many-to-many relationship was performed between class Person and Book using the intermediate class Rental.

### CODE REVIEW

File: classes/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'student'
+
+class Classroom
+  attr_accessor :label, :students
```
Comment (DeliceLydia) [inline classes/classroom.rb]: - [x] I suggest that you make `students` an `attr_reader` only

File: classes/student.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require_relative 'Person'
 
 class Student < Person
+  attr_accessor :classroom
```
Comment (DeliceLydia) [inline classes/student.rb]: - [x] Please make `classroom` an `attr_reader` we only want to write it

File: classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (DeliceLydia) [inline classes/book.rb]: - [ ] In the `book` class you should create a `rental` method and remember the `book` itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```ruby
def add_rental(person, date)
    Rental.new(date, person, self)
  end
```

File: classes/person.rb [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     super()
     @age = age
     @parent_permission = parent_permission
     @name = name
+    @rentals = []
   end
```
Comment (DeliceLydia) [inline classes/person.rb]: - [x] Please create a `rental` method and remember `the person` itself should be sent as a parameter to create the new rental achieving the 'has-many' association
```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @emarkees, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/classroom.rb (hunk 1 above)
Comment (emarkees) [reply classes/classroom.rb]: Thank you @DeliceLydia

File: classes/person.rb (hunk 4 above)
Comment (emarkees) [reply classes/person.rb]: done

**Review by OLIPLICHE — APPROVED** [review]
Hi @emarkees, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested by the previous review👏
- Created associations between classes 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**